docs/tutorials/google/spin_echoes.ipynb

```diff
@@ -0,0 +1,1347 @@
+{
```

Comment:
<p>I would imagine that <code>SynchronizeTerminalMeasurements</code> would be useful when we have multiple measurements as pushing them to the same last moment would be beneficial.</p><p> </p><p>In this case, since we only have a single measurement and it's the terminal operation on that qubit, pushing it at the end actually makes the circuit output worse and we could get a better fidelity by having it the first operation on the moment?</p><p> </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/quantumlib/Cirq/pull/4373/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='36'/>

Reply:
<p>Yeah aside from putting the measurement in the last moment, this circuit isn't the best to show off what <code>SynchronizeTerminalMeasurements</code>  does. My suggested fix is to have small circuits designed specifically to show what each optimizer does, e.g. a circuit with three staggered measurements fed into <code>SynchronizeTerminalMeasurements</code> that comes out with non-staggered measurements. I don't think doing this is in the spec for this tutorial though, so not sure on what action to take for this.</p><p> </p><p>(It is true that putting the ancilla measurement right at the start would give better fidelity, but that would be considered cheating for this benchmark.)</p>

Reply:
Hmm, I guess we can leave it as it is here. It at least helps highlight what's available and advance users can evaluate based on context whether to use the optimization or not.